csharp/src/Drivers/Apache/Thrift/Service/Rpc/Thrift/TDoubleColumn.cs

```diff
@@ -78,14 +79,19 @@ public TDoubleColumn DeepCopy()
                   var _list178 = await iprot.ReadListBeginAsync(cancellationToken);
                   length = _list178.Count;
 
-                  buffer = new byte[length * 8];
+                  buffer = new byte[length * DoubleSize];
                   var memory = buffer.AsMemory();
                   var typedMemory = Unsafe.As<Memory<byte>, Memory<long>>(ref memory).Slice(0, length);
                   iprot.Transport.CheckReadBytesAvailable(buffer.Length);
                   await transport.ReadExactlyAsync(memory, cancellationToken);
                   for (int _i179 = 0; _i179 < length; ++_i179)
                   {
+#if NET6_0_OR_GREATER
                     typedMemory.Span[_i179] = BinaryPrimitives.ReverseEndianness(typedMemory.Span[_i179]);
+#else
+                    long source = BinaryPrimitives.ReverseEndianness(BitConverter.ToInt64(buffer, _i179 * DoubleSize));
+                    BitConverter.GetBytes(source).CopyTo(buffer, _i179 * DoubleSize);
```

Comment:
This is pretty expensive because it needs to allocate an array. Consider doing something with unsafe code.

Reply:
Alternatively, just swap the bytes manually. We already have a non-optimal implementation of ReverseEndianI32AtOffset in StreamExtensions.cs. It would be trivial to create a similar ReverseEndianI64AtOffset and use it for both TDoubleColumn and T64Column.

Reply:
@CurtHagenlocher 
Just an explanation, it seems the behavior of the Unsafe implementation is different before NET 6 - hence the conditional compilation. I'll attempt a more efficient implementation.

Reply:
Done.